stripe/src/test/java/com/stripe/android/StripeKotlinTest.kt

```diff
@@ -0,0 +1,545 @@
+package com.stripe.android
+
+import com.nhaarman.mockitokotlin2.any
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.whenever
+import com.stripe.android.exception.APIException
+import com.stripe.android.exception.AuthenticationException
+import com.stripe.android.exception.InvalidRequestException
+import com.stripe.android.model.PaymentIntent
+import com.stripe.android.model.SetupIntent
+import com.stripe.android.model.Source
+import com.stripe.android.networking.ApiRequest
+import com.stripe.android.networking.StripeApiRepository
+import kotlinx.coroutines.runBlocking
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+import kotlin.test.assertFailsWith
+import kotlin.test.assertSame
+
+/**
+ * Test for [Stripe] suspend functions.
+ */
+@RunWith(JUnit4::class)
+internal class StripeKotlinTest {
+    private val mockApiRepository: StripeApiRepository = mock()
```

Comment:
You might consider using a fake instead of a mock. You can extend `AbsFakeStripeRepository` and implement only the desired methods. You can also trigger an error when needed.

Reply:
Trying to limit the scope of mock/fake in this test, since we're not calling any stateful function from `StripeRespository` and the only broken point is when `StripeRespository` throws certain `Exception` or returns null, I think mocking it should be enough, can change this to fake later if the logic becomes more complicated